src/handlers/notify_zulip.rs

```diff
@@ -1,96 +1,171 @@
 use crate::{
-    config::NotifyZulipConfig,
-    github::{IssuesAction, IssuesEvent},
+    config::{NotifyZulipConfig, NotifyZulipLabelConfig},
+    github::{Issue, IssuesAction, IssuesEvent, Label},
     handlers::Context,
 };
 
 pub(super) struct NotifyZulipInput {
     notification_type: NotificationType,
+    /// Label that triggered this notification.
+    ///
+    /// For example, if an `I-prioritize` issue is closed,
+    /// this field will be `I-prioritize`.
+    label: Label,
 }
 
 pub(super) enum NotificationType {
     Labeled,
     Unlabeled,
+    Closed,
+    Reopened,
 }
 
 pub(super) fn parse_input(
     _ctx: &Context,
     event: &IssuesEvent,
     config: Option<&NotifyZulipConfig>,
-) -> Result<Option<NotifyZulipInput>, String> {
-    if let IssuesAction::Labeled | IssuesAction::Unlabeled = event.action {
-        let applied_label = &event.label.as_ref().expect("label").name;
-        if let Some(config) = config.and_then(|c| c.labels.get(applied_label)) {
-            for label in &config.required_labels {
-                let pattern = match glob::Pattern::new(label) {
-                    Ok(pattern) => pattern,
-                    Err(err) => {
-                        log::error!("Invalid glob pattern: {}", err);
-                        continue;
-                    }
-                };
-                if !event
-                    .issue
-                    .labels()
-                    .iter()
-                    .any(|l| pattern.matches(&l.name))
-                {
-                    // Issue misses a required label, ignore this event
-                    return Ok(None);
+) -> Result<Option<Vec<NotifyZulipInput>>, String> {
+    let config = match config {
+        Some(config) => config,
+        None => return Ok(None),
+    };
+
+    match event.action {
+        IssuesAction::Labeled | IssuesAction::Unlabeled => {
+            let applied_label = event.label.as_ref().expect("label").clone();
```

Comment:
This is probably better, right?

```suggestion
            let applied_label = event.label.cloned().expect("label");
```

Reply:
I personally find it cleaner